### PR TITLE
Hw29

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/InjectDataController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            throws ServletException, IOException {
+        User userJoe = new User("Joe", "joe_hann", "meteora");
+        userJoe.setRoles(List.of(Role.of("USER")));
+        User userChes = new User("Chester", "ches_benning", "hybrid");
+        userChes.setRoles(List.of(Role.of("USER")));
+        User admin = new User("Admin", "lp", "admin");
+        admin.setRoles(List.of(Role.of("ADMIN")));
+        userService.create(userJoe);
+        userService.create(userChes);
+        userService.create(admin);
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/controllers/InjectDataController.java:29]: maybe you should add shopping cart to Joe and Chester

File: src/main/java/com/internet/shop/model/User.java [hunk 2]
```diff
@@ -1,10 +1,13 @@
 package com.internet.shop.model;
 
+import java.util.List;
+
 public class User {
     private Long userId;
     private String name;
     private String login;
     private String password;
+    private List<Role> roles;
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/model/User.java]: let's make this field - Set
```suggestion
    private Set<Role> roles;
```

File: src/main/java/com/internet/shop/model/User.java [hunk 3]
```diff
@@ -44,6 +47,14 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    public List<Role> getRoles() {
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/model/User.java]:
```suggestion
    public Set<Role> getRoles() {
```

File: src/main/java/com/internet/shop/model/User.java [hunk 4]
```diff
@@ -44,6 +47,14 @@ public void setPassword(String password) {
...
+        return roles;
+    }
+
+    public void setRoles(List<Role> roles) {
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/model/User.java]:
```suggestion
    public void setRoles(Set<Role> roles) {
```

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+        if (protectedUrls.get(requestedUrl) == null) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null || userService.get(userId) == null) {
+            resp.sendRedirect("/login");
+            return;
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]: we did this check in Authentification Filter
```suggestion

```

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        User user = userService.get(userId);
+
+        if (isAuthorized(user, protectedUrls.get(requestedUrl))) {
+            filterChain.doFilter(req, resp);
+            return;
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]: this `return` is redundant

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        } else {
+            req.getRequestDispatcher("/WEB-INF/views/user/accessDenied.jsp").forward(req, resp);
+            return;
```
Comment (KovalAlexey) [inline src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]: same here

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java (hunk 5 above)
Comment (margarita-ignatieva) [reply src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]: forgot to make changes in Authentication filter, thanks

File: src/main/java/com/internet/shop/controllers/InjectDataController.java (hunk 1 above)
Comment (margarita-ignatieva) [reply src/main/java/com/internet/shop/controllers/InjectDataController.java:29]: They are actually redundant, it's mainly for Admin usage injection

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorisationFilter implements Filter {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]:
```suggestion
public class AuthorizationFilter implements Filter {
```

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        protectedUrls.put("/orders/manage", List.of(Role.RoleName.USER));
+        protectedUrls.put("/orders/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("shopping-cart/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/admin/details", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/all", List.of(Role.RoleName.ADMIN));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]: i don't see user orders and user order details

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        if (isAuthorized(user, protectedUrls.get(requestedUrl))) {
+            filterChain.doFilter(req, resp);
+        } else {
+            req.getRequestDispatcher("/WEB-INF/views/user/accessDenied.jsp").forward(req, resp);
+        }
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java (hunk 9 above)
Comment (margarita-ignatieva) [reply src/main/java/com/internet/shop/web/filters/AuthorisationFilter.java]: missed this, added

File: src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+        protectedUrls.put("shopping-cart/products/delete", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products/add", List.of(Role.RoleName.USER));
+        protectedUrls.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/users/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/manage", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/products/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders/admin/details", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/orders", List.of(Role.RoleName.ADMIN));
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java:42]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**